### ISSUE TITLE
“URI” doesn’t match RFC 3986

### ISSUE THREAD

**Issue by FGasper** [wamp-proto/wamp-proto#276]
It occurs to me: why does the WAMP spec use the term “URI” to describe strings that don’t have a scheme?

Examples from RFC 3986:

- ```ftp://ftp.is.co.za/rfc/rfc1808.txt```
- ```ldap://[2001:db8::7]/c=GB?objectClass?one```
- ```mailto:John.Doe@example.com```
- ```news:comp.infosystems.www.servers.unix```
- ```tel:+1-816-555-1212```
- ```telnet://192.0.2.16:80/```
- ```urn:oasis:names:specification:docbook:dtd:xml:4.1.2```

… yet WAMP uses strings like:

- ```wamp.error.not_authorized```
- ```wamp.error.procedure_already_exists```

Am I missing something? It seems like WAMP is actually redefining the term “URI”, given that the examples in the protocol would not satisfy the definition in the referenced RFC.

**Comment by oberstet** [comment]
Yes, this should be brought in line with [RFC 3986](https://tools.ietf.org/html/rfc3986) and [RFC 3987](https://tools.ietf.org/html/rfc3987).

I think a light weight way would be do state that we use eg `wamp.error.not_authorized` as an abbreviation for `wamp:wamp.error.not_authorized`. That is, omit a `wamp:` scheme prefix, whenever there is no ambiguity (eg on the wire when 2 WAMP peers are talking to each other).

In general, the idea of referring the RFCs for WAMP was to free ride on established specs of "IDs", and possibly also free ride on implementations for processing such IDs, eg from RFC3986 

> It thus defines the syntax and semantics needed to implement a scheme-
> independent parsing mechanism for URI references, by which the
> scheme-dependent handling of a URI can be postponed until the
> scheme-dependent semantics are needed.

---

If we do so, we should also register the scheme:

> The process for registration of new URI schemes is defined separately by [BCP35].

A `wamp:` scheme is not registered yet: http://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml#uri-schemes-3

> WAMP has the WebSocket subprotocol identifier `wamp` [properly registered](https://www.iana.org/assignments/websocket/websocket.xhtml) with the IETF, but not anything else ..
